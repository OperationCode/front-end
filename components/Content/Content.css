.Content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 1rem auto;
}

.title {
  text-align: center;
}

.underline {
  border-bottom: 4px solid var(--primary);
  margin-bottom: 1rem;
}

.columnsContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

/* the individual columns */
.columnsContainer > * {
  margin: 1rem;
}
