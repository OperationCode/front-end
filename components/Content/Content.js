import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Container from 'components/_common_/Container/Container';
import styles from './Content.css';

Content.propTypes = {
  backgroundImageSource: PropTypes.string,
  columns: PropTypes.array.isRequired, // can be JSX, elements, or strings
  hasTitleUnderline: PropTypes.bool,
  id: PropTypes.string,
  isFullViewportHeight: PropTypes.bool,
  theme: PropTypes.string,
  title: PropTypes.string,
};

Content.defaultProps = {
  backgroundImageSource: undefined,
  hasTitleUnderline: false,
  id: undefined,
  isFullViewportHeight: false,
  theme: 'secondary',
  title: undefined,
};

function Content({
  columns,
  hasTitleUnderline,
  id,
  isFullViewportHeight,
  theme,
  title,
  backgroundImageSource,
}) {
  return (
    <Container
      backgroundImageSource={backgroundImageSource}
      id={id}
      isFullViewportHeight={isFullViewportHeight}
      theme={theme}
    >
      {title && (
        <h3
          className={classNames(styles.title, {
            [styles.underline]: hasTitleUnderline,
          })}
        >
          {title}
        </h3>
      )}

      <div className={styles.columnsContainer}>
        {/* eslint-disable-next-line react/no-array-index-key */}
        {columns.map((column, index) => React.cloneElement(column, { key: index }))}
      </div>
    </Container>
  );
}

export default Content;
