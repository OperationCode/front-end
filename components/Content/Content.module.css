.Content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 1rem auto;
}

.columnsContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

/* the individual columns */
.columnsContainer > * {
  margin: 1rem;
}
