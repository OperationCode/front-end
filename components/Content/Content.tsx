import { cloneElement, ReactElement } from 'react';
import Container from 'components/Container/Container';
import Heading from 'components/Heading/Heading';

export type ContentPropsType = {
  columns: React.ReactNode[];
  backgroundImageSource?: string;
  className?: string;
  hasTitleUnderline?: boolean;
  id?: string;
  isFullViewportHeight?: boolean;
  theme?: 'gray' | 'secondary' | 'white';
  title?: string;
};

function Content({
  className,
  columns,
  hasTitleUnderline = false,
  id,
  isFullViewportHeight = false,
  theme = 'secondary',
  title,
  backgroundImageSource,
}: ContentPropsType) {
  return (
    <Container
      backgroundImageSource={backgroundImageSource}
      className={className}
      id={id}
      isFullViewportHeight={isFullViewportHeight}
      theme={theme}
    >
      {title && <Heading text={title} hasTitleUnderline={hasTitleUnderline} headingLevel={3} />}

      <div className="flex justify-center items-center flex-wrap w-full [&>*]:m-4">
        {/* eslint-disable-next-line react/no-array-index-key */}
        {columns.map((column, index) => cloneElement(column as ReactElement<any>, { key: index }))}
      </div>
    </Container>
  );
}

export default Content;
