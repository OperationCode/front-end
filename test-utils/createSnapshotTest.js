/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
