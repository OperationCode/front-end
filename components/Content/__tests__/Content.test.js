import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Content from '../Content';

// eslint-disable-next-line unicorn/prevent-abbreviations
const requiredProps = {
  columns: [
    <div>
      <p>some test content</p>
    </div>,
  ],
  id: undefined,
};

describe('Content', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Content {...requiredProps} />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Content
        columns={[
          <div>
            <p>some multi-column test content</p>
          </div>,
          <h6>OH YEAH!!</h6>,
          <aside>Testing...</aside>,
        ]}
        hasTitleUnderline
        id="test-id"
        isFullViewportHeight
        title="Testing!"
      />,
    );
  });
});
