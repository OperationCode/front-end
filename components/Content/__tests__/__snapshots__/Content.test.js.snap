// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render with many props assigned 1`] = `
<div
  className="Container secondary fullViewportHeight"
  id="test-id"
>
  <div
    className="content"
  >
    <div
      className="headingContainer"
    >
      <h3
        className="Heading underline"
      >
        <div
          className="hashLinkContainer"
          data-testid="Heading Content testing-link"
        >
          <a
            data-testid="Hash Link"
            href="#testing-link"
            id="testing-link"
          >
            <span
              className="ScreenReaderOnly"
              data-testid="SCREEN_READER_ONLY"
            >
              Scroll Link for 
              Testing!
            </span>
            <svg
              className="icon"
            />
          </a>
          Testing!
        </div>
      </h3>
    </div>
    <div
      className="columnsContainer"
    >
      <div>
        <p>
          some multi-column test content
        </p>
      </div>
      <h6>
        OH YEAH!!
      </h6>
      <aside>
        Testing...
      </aside>
    </div>
  </div>
</div>
`;

exports[`Content should render with required props 1`] = `
<div
  className="Container secondary"
>
  <div
    className="content"
  >
    <div
      className="columnsContainer"
    >
      <div>
        <p>
          some test content
        </p>
      </div>
    </div>
  </div>
</div>
`;
